src/laser/MultiLaser.cpp

```diff
@@ -243,44 +254,79 @@ MultiLaser::GetEnvelopeFromFile (const amrex::Box& domain) {
-    //lasy: tyx in C order
+    // Extract grid offset and grid spacing from laser file
+    std::vector<double> offset = laser.gridGlobalOffset();
+    std::vector<double> position = laser_comp.position<double>();
+    std::vector<double> spacing = laser.gridSpacing<double>();
 
-    if (static_cast<std::uint64_t>(domain.length(0)) != extend[2] ||
-        static_cast<std::uint64_t>(domain.length(1)) != extend[1] ||
-        static_cast<std::uint64_t>(domain.length(2)) != extend[0]) {
-        amrex::Abort("Incompatible box sizes. HiPACE++ (" +
-        std::to_string(domain.length(0)) + ", " + std::to_string(domain.length(1)) + ", "
-        + std::to_string(domain.length(2)) + "), "
-        + "File (" + std::to_string(extend[2]) + ", " + std::to_string(extend[1]) + ", "
-        + std::to_string(extend[0]) + ")\n");
-    }
+    // Calculate the min and max of the grid from laser file
+    amrex::Real tmin_laser = offset[0] + position[0]*spacing[0];
+    amrex::Real ymin_laser = offset[1] + position[1]*spacing[1];
+    amrex::Real xmin_laser = offset[2] + position[2]*spacing[2];
+    AMREX_ALWAYS_ASSERT(position[0] == 0 && position[1] == 0 && position[2] == 0);
 
-    amrex::Dim3 arr_begin = {domain.smallEnd(0), domain.smallEnd(1), domain.smallEnd(2)};
-    amrex::Dim3 arr_end = {domain.smallEnd(0) + domain.length(0),
-                           domain.smallEnd(1) + domain.length(1),
-                           domain.smallEnd(2) + domain.length(2)};
+    //lasy: tyx in C order
+    amrex::Dim3 arr_begin = {0, 0, 0};
+    amrex::Dim3 arr_end = {static_cast<int>(extent[2]), static_cast<int>(extent[1]), static_cast<int>(extent[0])};
     amrex::Array4<input_type> input_file_arr(data.get(), arr_begin, arr_end, 1);
+
+    //hipace: xyt in Fortran order
     amrex::Array4<amrex::Real> laser_arr = m_F_input_file.array();
-    const int lo_t = domain.smallEnd(2);
-    const int hi_t = domain.bigEnd(2);
 
     series.flush();
 
+    constexpr int interp_order_xy = 1;
+    const amrex::Real dx = gm.CellSize(Direction::x);
+    const amrex::Real dy = gm.CellSize(Direction::y);
+    const amrex::Real dz = gm.CellSize(Direction::z);
+    const amrex::Real xmin = gm.ProbLo(Direction::x)+dx/2;
+    const amrex::Real ymin = gm.ProbLo(Direction::y)+dy/2;
+    const amrex::Real zmin = gm.ProbLo(Direction::z)+dz/2;
+    const amrex::Real zmax = gm.ProbHi(Direction::z)-dz/2;
+
     for (int k = domain.smallEnd(2); k <= domain.bigEnd(2); ++k) {
         for (int j = domain.smallEnd(1); j <= domain.bigEnd(1); ++j) {
             for (int i = domain.smallEnd(0); i <= domain.bigEnd(0); ++i) {
-                laser_arr(i, j, k, 0) = static_cast<amrex::Real>(
-                    input_file_arr(i, j, lo_t + hi_t - k).real() * unitSI
-                );
-                laser_arr(i, j, k, 1) = static_cast<amrex::Real>(
-                    input_file_arr(i, j, lo_t + hi_t - k).imag() * unitSI
-                );
+
+                const amrex::Real x = i*dx + xmin;
```

Comment:
This assumes that `domain.smallEnd(0)` is zero.

Reply:
I just fixed it with `i-imin` instead of `i` etc.